src/main/java/com/karlofduty/NoVoidLoss/DeathListener.java

```diff
@@ -1,61 +1,21 @@
 package com.karlofduty.NoVoidLoss;
 
-import org.bukkit.Material;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
-import org.bukkit.event.entity.EntityDamageEvent;
 import org.bukkit.event.entity.PlayerDeathEvent;
-import org.bukkit.event.player.PlayerRespawnEvent;
-import org.bukkit.inventory.ItemStack;
-
-import java.util.Map;
 
 public class DeathListener implements Listener
 {
-    private NoVoidLoss plugin;
-
-    public DeathListener(final NoVoidLoss plugin)
-    {
-        this.plugin = plugin;
-    }
-
-    @EventHandler(priority = EventPriority.HIGHEST)
-    public void onEvent(final PlayerDeathEvent event)
-    {
-        boolean restoreAllItems = false;
-        try {
-            restoreAllItems = event.getEntity().getLastDamageCause().getCause() == EntityDamageEvent.DamageCause.VOID
```

Comment:
This rewrite loses one of the death reasons.

Reply:
If a player dies due to suffocation damage, they wouldn't be inside the void and they'd be able to run back and recover their items.

The only real server related case I can see is when the server crashes, and the player gets stuck inside a block due to a rollback. In this case the player can just use a pickaxe to dig themselves out.

Reply:
Players should still be protected in suffocation deaths.